### PR TITLE
Istio: Upgrade to V1 of APIs

### PR DESCRIPTION
**Pull request by evenh**
See https://istio.io/latest/blog/2024/v1-apis/

## Summary by CodeRabbit

- **New Features**
	- Introduced new `NetworkPolicy` resources and updated existing ones for better traffic management.
	- Added `ServiceAccount`, `Deployment`, `HorizontalPodAutoscaler`, and `Service` resources for enhanced application deployment and scaling.
	- Added `SKIPJob` resources to manage job specifications within the application.

- **Improvements**
	- Updated API versions for `PeerAuthentication`, `ServiceEntry`, `Gateway`, and `VirtualService` resources to stable `v1` versions, ensuring compatibility with the latest Istio standards.
	- Enhanced `ServiceEntry` configurations to define external services with updated API versions.

- **Bug Fixes**
	- Corrected formatting and indentation issues in YAML files for improved readability and structure.

### CODE REVIEW

**Review by BardOve — APPROVED** [review]

**Review by Griznah — APPROVED** [review]

File: internal/controllers/application.go [hunk 1]
```diff
@@ -83,18 +84,18 @@ func (r *ApplicationReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+		Owns(&istionetworkingv1.ServiceEntry{}).
+		Owns(&istionetworkingv1.Gateway{}, builder.WithPredicates(
+			util.MatchesPredicate[*istionetworkingv1.Gateway](isIngressGateway),
 		)).
 		Owns(&telemetryv1.Telemetry{}).
 		Owns(&autoscalingv2.HorizontalPodAutoscaler{}).
-		Owns(&networkingv1beta1.VirtualService{}).
-		Owns(&securityv1beta1.PeerAuthentication{}).
+		Owns(&istionetworkingv1.VirtualService{}).
+		Owns(&securityv1.PeerAuthentication{}).
```
[the remark this answers is not in the document]
Comment (evenh) [reply internal/controllers/application.go:94]: There is no v1 of types: https://github.com/istio/api/tree/1.23.3/type

File: internal/controllers/skipjob.go [hunk 2]
```diff
@@ -15,7 +16,7 @@ import (
 	"github.com/kartverket/skiperator/pkg/resourcegenerator/podmonitor"
 	"github.com/kartverket/skiperator/pkg/resourcegenerator/resourceutils"
 	"github.com/kartverket/skiperator/pkg/resourcegenerator/serviceaccount"
-	istionetworkingv1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
+	istionetworkingv1 "istio.io/client-go/pkg/apis/networking/v1"
```
[the remark this answers is not in the document]
Comment (evenh) [reply internal/controllers/skipjob.go:19]: N/A

File: pkg/resourcegenerator/istio/telemetry/telemetry.go [hunk 3]
```diff
@@ -41,7 +42,7 @@ func Generate(r reconciliation.Reconciliation) error {
 	}
 	telemetry.Spec = telemetryapiv1.Telemetry{
 		Tracing: telemetryTracing,
-		Selector: &v1beta1.WorkloadSelector{
+		Selector: &typev1beta1.WorkloadSelector{
 			MatchLabels: object.GetDefaultLabels(),
 		},
```
[the remark this answers is not in the document]
Comment (evenh) [reply pkg/resourcegenerator/istio/telemetry/telemetry.go:47]: N/A

File: pkg/resourcegenerator/istio/peerauthentication/peer_authentication.go [hunk 4]
```diff
@@ -24,14 +25,14 @@ func Generate(r reconciliation.Reconciliation) error {
...
 	ctxLog.Debug("Attempting to generate peer authentication for application", "application", application.Name)
 
-	peerAuthentication := securityv1beta1.PeerAuthentication{ObjectMeta: metav1.ObjectMeta{Namespace: application.Namespace, Name: application.Name}}
+	peerAuthentication := securityv1.PeerAuthentication{ObjectMeta: metav1.ObjectMeta{Namespace: application.Namespace, Name: application.Name}}
 
-	peerAuthentication.Spec = securityv1beta1api.PeerAuthentication{
+	peerAuthentication.Spec = securityv1api.PeerAuthentication{
 		Selector: &typev1beta1.WorkloadSelector{
 			MatchLabels: util.GetPodAppSelector(application.Name),
 		},
```
[the remark this answers is not in the document]
Comment (evenh) [reply pkg/resourcegenerator/istio/peerauthentication/peer_authentication.go:33]: N/A

File: pkg/resourcegenerator/istio/peerauthentication/peer_authentication.go [hunk 5]
```diff
@@ -2,12 +2,13 @@ package peerauthentication
 
 import (
 	"fmt"
+
 	skiperatorv1alpha1 "github.com/kartverket/skiperator/api/v1alpha1"
 	"github.com/kartverket/skiperator/pkg/reconciliation"
 	"github.com/kartverket/skiperator/pkg/util"
-	securityv1beta1api "istio.io/api/security/v1beta1"
+	securityv1api "istio.io/api/security/v1"
 	typev1beta1 "istio.io/api/type/v1beta1"
```
[the remark this answers is not in the document]
Comment (evenh) [reply pkg/resourcegenerator/istio/peerauthentication/peer_authentication.go:10]: N/A

**Comment by evenh** [comment]
@coderabbitai Please generate functionality and tests for transitioning from the Istio v1beta1 to v1 API's, ensuring that existing generated resources using v1beta1 won't be dangling when skiperator generates v1 resources.

**Comment by evenh** [comment]
After diving into this problem, I recalled that multiple versions of a CRD are basically different views of one set of data. When there's no changes (like the convertion from `v1beta1` to `v1` API's), no change should be required on our end. The outlier is the `Telemetry` API, but we started with `v1` so we should be good there as well.

See [K8s CRD versioning](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning) and this blog post from Istio's John Howard on [their approach to CRD versioning](https://blog.howardjohn.info/posts/crd-versioning/).

To sum it up; this PR should be mergable in its current state.

**Pull request merged**